jax/random.py

```diff
@@ -374,20 +374,52 @@ def uniform(key: jnp.ndarray,
     dtype: optional, a float dtype for the returned values (default float64 if
       jax_enable_x64 is true, otherwise float32).
     minval: optional, a minimum (inclusive) value broadcast-compatible with shape for the range (default 0).
-    maxval: optional, a maximum (exclusive) value broadcast-compatible with shape for the range (default 1).
+    maxval: optional, a maximum (exclusive) value broadcast-compatible with shape for the range (default 1 if not complex, otherwise 1+1j).
 
   Returns:
     A random array with the specified shape and dtype.
   """
-  if not dtypes.issubdtype(dtype, np.floating):
+  if minval is None:
+    minval = _uniform_minval_default(dtype)
+  if maxval is None:
+    maxval = _uniform_maxval_default(dtype)
+
+  if not (jnp.issubdtype(dtype, np.floating) or jnp.issubdtype(dtype, np.complexfloating)):
     raise ValueError(f"dtype argument to `uniform` must be a float dtype, "
                      f"got {dtype}")
   dtype = dtypes.canonicalize_dtype(dtype)
   shape = abstract_arrays.canonicalize_shape(shape)
   return _uniform(key, shape, dtype, minval, maxval)  # type: ignore
 
+def _uniform_minval_default(dtype: np.dtype) -> [float, complex, jnp.ndarray]:
```

Comment:
This one feels redundant, in the sense that you could have just written:
`np.array(0, dtype=dtype)`